src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -98,6 +105,238 @@ public async ValueTask DisposeAsync()
+        }
+    }
+
+    private static (string NetCoreVersion, string AspNetCoreVersion) GetFrameworkVersionsFromRuntimeConfig(string assemblyPath)
+    {
+        // Find the AppHost's runtimeconfig.json file
+        string runtimeConfigPath;
+        if (string.Equals(".dll", Path.GetExtension(assemblyPath), StringComparison.OrdinalIgnoreCase))
+        {
+            runtimeConfigPath = Path.ChangeExtension(assemblyPath, ".runtimeconfig.json");
+        }
+        else
+        {
+            // For executables, the runtime config is named after the base executable name
+            // Handle both Windows (.exe) and Unix (no extension) executables
+            var directory = Path.GetDirectoryName(assemblyPath)!;
+            var fileName = Path.GetFileName(assemblyPath);
+            var baseName = Path.GetExtension(fileName) switch
+            {
+                ".exe" => Path.GetFileNameWithoutExtension(fileName), // Windows: remove .exe
+                _ => fileName // Unix or other: use full filename as base
+            };
+            runtimeConfigPath = Path.Combine(directory, $"{baseName}.runtimeconfig.json");
+        }
+
+        if (!File.Exists(runtimeConfigPath))
+        {
+            // Fallback to runtime detection if runtime config doesn't exist
+            return GetFallbackFrameworkVersions();
+        }
+
+        // Parse the AppHost's runtime config to get framework versions
+        var configText = File.ReadAllText(runtimeConfigPath);
+        var configJson = JsonNode.Parse(configText)?.AsObject();
+
+        if (configJson is null)
+        {
+            throw new DistributedApplicationException($"Failed to parse AppHost runtime config: {runtimeConfigPath}");
+        }
+
+        string netCoreVersion = FallbackNetCoreVersion; // Default fallback
+        string aspNetCoreVersion = FallbackAspNetCoreVersion; // Default fallback
+
+        if (configJson["runtimeOptions"]?.AsObject() is { } runtimeOptions &&
+            runtimeOptions["frameworks"]?.AsArray() is { } frameworks)
+        {
+            foreach (var framework in frameworks)
+            {
+                if (framework?.AsObject() is { } frameworkObj &&
+                    frameworkObj["name"]?.GetValue<string>() is { } name &&
+                    frameworkObj["version"]?.GetValue<string>() is { } version)
+                {
+                    switch (name)
+                    {
+                        case "Microsoft.NETCore.App":
+                            netCoreVersion = version;
+                            break;
+                        case "Microsoft.AspNetCore.App":
+                            aspNetCoreVersion = version;
+                            break;
+                    }
+                }
+            }
+        }
+
+        return (netCoreVersion, aspNetCoreVersion);
+    }
+
+    private static (string NetCoreVersion, string AspNetCoreVersion) GetFallbackFrameworkVersions()
```

Comment:
@DamianEdwards why isn't this just returning the framework FallbackAspNetCoreVersion and FallbackNetCoreVersion?